### PR TITLE
Use a LinkedBlockingQueue instead of polling.

### PR DESCRIPTION
**Pull request by zainab-ali**
This resolves [the busy spin in SBT](https://github.com/disneystreaming/weaver-test/issues/671).

We use a [LinkedBlockingQueue](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/LinkedBlockingQueue.html) and a blocking poll operation to poll every second.

### CODE REVIEW

File: modules/framework/jvm/src/main/scala/SbtTask.scala [hunk 1]
```diff
@@ -30,7 +31,7 @@ private[framework] class SbtTask(
     loggerPermit.acquire()
     try {
       while (!finished && !isDone.get()) {
-        val nextEvent = Option(queue.poll())
+        val nextEvent = Option(queue.poll(1, TimeUnit.SECONDS))
```
Comment (zainab-ali) [inline modules/framework/jvm/src/main/scala/SbtTask.scala:34]: The 1 second time is arbitrary.

The loop should terminate when `isDone` is set, so probably shouldn't poll indefinitely. From reading the code, `isDone` ought to be set whenever there's an error creating global resources or whenever SBT calls `done` on the runner. I haven't been able to simulate either of these cases.

Comment (Baccata) [reply modules/framework/jvm/src/main/scala/SbtTask.scala:34]: I think 1 second is fine. 

It's really hard to test a test framework via the SBT interface, so as long as the build still builds, I think your changes are fine 👍

Comment (zainab-ali) [reply modules/framework/jvm/src/main/scala/SbtTask.scala:34]: I tested with scala-cli and SBT, and it seems to work fine.

You can reproduce the spinlock with 
```
scala-cli test https://gist.github.com/zainab-ali/867fbc095d2ab3ceda498b29bc2b8aba
```
Here's a screenshot of the CPU usage before:
before

And after (with the current changes):
after

Comment (Baccata) [reply modules/framework/jvm/src/main/scala/SbtTask.scala:34]: Awesome, that's sufficient to get it merged then 👍

**Review by Baccata — APPROVED** [review]

**Pull request merged**